File: real_estate-app/src/app/models/Rezervimi.ts
import { ShtepiaProfile } from "./ShtepiaProfile";
import { UserProfile } from "./userProfile";




export interface Rezervimi{
    rezervimiId: string;
    check_in: string;
    check_out: string;
    nrPersonave: string;
    pagesa: string;
    kontrata:string;
    appUserId?: string;
    user?: UserProfile;


}
// export interface Rezervimi {
//     rezervimiId: string;
//     // nrRezervimit?:string;
//     check_in: string;
//     check_out: string;
//     // shtepiaId?: string;
//     // menyraPagesesId?: string;
//     // kontrataId?: string;
//     IsCancelled?:boolean;
//     AppUserId?:string;
//     user?: UserProfile;
//     Attendeess?: RezervimiAttendees[]
    
// }



// export interface RezervimiAttendees{
//     id?: string;
 
//     RezervimiId?: string;
//     Cmimi: string;
//     IsHost?: boolean;
// }

File: Domain/Rezervimi.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;

using System.Threading.Tasks;

namespace Domain
{
    public class Rezervimi
    {
         
  //       public Guid RezervimiId { get; set; }

  //               // public string nrRezervimit { get; set; } = string.Empty;

  //       public DateTime Check_in { get; set; }

  //       public DateTime Check_out { get; set; }

  //     //   public string AppUserId { get; set; }
  //     //      [ForeignKey("AppUserId")]
  //     //   public AppUser AppUser { get; set; }
  //  //      public Guid? ShtepiaId {get;set;}
  //  //         [ForeignKey("ShtepiaId")]
  //  //      public Shtepia Shtepia {get;set;}

  //  //       public Guid? MenyraPagesesId { get; set; }
  //  //   [ForeignKey("MenyraPagesesId")]

  //  //    public MenyraPageses MenyraPageses { get; set; }

  //  //    public Guid? KontrataId{get;set;}
  //  //       [ForeignKey("KontrataId")]
  //  //    public Kontrata Kontrata{get;set;}
      
  //     // public bool IsHost {get;set;}

  //   //   public bool IsCancelled{get;set;}
       
  //      public Guid AppUserId { get; set; } 
  //       public AppUser? AppUser { get; set; }
  //   //    public ICollection<RezervimiAttendee> Attendees{get; set;}=new List<RezervimiAttendee>();

      
     
  //   }

    
        public Guid RezervimiId { get; set; }
        public DateTime Check_in { get; set; }
        public DateTime Check_out { get; set; }
        public string nrPersonave { get; set; }
        public string Pagesa { get; set; }
        public Guid AppUserId { get; set; } 
        public AppUser AppUser { get; set; }

        
    }
}